File: 7. Reverse Integer/solution.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
  const isNeg = 0 > x ? -1 : 1;
  let result = 0;
  let absx = Math.abs(x);

  while (absx > 0) {
    result = result * 10 + (absx % 10);
    absx = Math.floor(absx / 10);
  }

  if (result < Math.pow(-2, 31) || result > Math.pow(2, 31) - 1) return 0;

  return result * isNeg;
};

File: 7. Reverse Integer/solution.go
import "math"

func reverse(x int) int {
	var result = 0
	var isNegative = 1

	if x < 0 {
		x *= -1
		isNegative *= -1
	}

	for x != 0 {
		result = result*10 + x%10
		x /= 10
	}

	if math.MinInt32 > result || math.MaxInt32 < result {
		result = 0
	}

	return result * isNegative
}
